DBAgenda.php

```diff
@@ -0,0 +1,311 @@
+<?php
+
+use Monolog\Logger;
+use Sabre\VObject;
+
+abstract class DBAgenda extends Agenda {
+    protected $pdo;
+    
+    public function __construct(string $url, string $username, string $password, object $api, array $agenda_args) {
+        setLogHandlers($this->log);
+        
+        $this->caldav_client = new CalDAVClient($url, $username, $password);
+        $this->api = $api;
+
+        $this->openDB($agenda_args);
+        
+        $this->pdo->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+        $this->pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION); // ERRMODE_WARNING | ERRMODE_EXCEPTION | ERRMODE_SILENT
+
+    }
+    
+    abstract public function createDB();
+    abstract protected function openDB(array $agenda_args);
+    abstract protected function getLastInsertedRowId();
+    
+    public function getUserEventsFiltered(string $userid, array $filters_to_apply = array()) {
+        $sql = 'SELECT vCalendarFilename, number_volunteers_required, vCalendarRaw FROM events WHERE ';
+        $sql .= 'Date(datetime_begin) > :datetime_begin ';
+        
+        $intersect = array();
+        if(isset($filter["my_events"])) {
+            $intersect[] = "SELECT vCalendarFilename FROM events_attendees
+INNER JOIN attendees
+WHERE attendees.email = events_attendees.email and attendees.userid = '$userid'";
```

Comment:
I'm not sure I understand why the WHERE is both on the email and the userid: don't we assume a 1 <-> 1 relation between the two?

Reply:
I am not an SQL expert but it appears that:
`SELECT event_filename FROM events_attendees
INNER JOIN attendees
WHERE attendees.userid = "$userid";`
Is equivalent to :
`SELECT event_filename FROM events_attendees;`
We need to explicitly indicate the relationship between the tables `events_attendees` and `attendees`.
Foreign keys are not "auto connected".

Reply:
ah, the one on the email is for the join, and the one on userid is to get only the user we're interested in. Ok, I missed that. Thanks for the explanation!